src/runtime.ts

```diff
@@ -602,3 +621,48 @@ export async function createDefaultRuntime(
     ...(underlyingAccount && { underlyingAccount }),
   };
 }
+
+/**
+ * Creates and switches to a new runtime for identity.
+ *
+ * @param      {Runtime}  existingRuntime  existing runtime instance
+ * @param      {string}   identity         identity address
+ * @return     {Promise<Runtime>}  runtime instance
+ */
+export async function getRuntimeForIdentity(existingRuntime: Runtime,
+  identity: string): Promise<Runtime> {
+  const identityList = await existingRuntime.profile.getIdentityAccessList();
+  let key = identityList[identity]?.identityAccess;
+  // check if identityList contains the idenitity, if not throw an error.
+  if (!key) {
+    const addressHash = existingRuntime.nameResolver.soliditySha3(
+      ...[
+        existingRuntime.nameResolver.soliditySha3(identity),
+        existingRuntime.nameResolver.soliditySha3(existingRuntime.activeIdentity),
+      ].sort(),
+    );
+    key = identityList[addressHash]?.identityAccess;
+    if (!key) {
+      throw new Error('Access to identity is not permitted');
+    }
+  }
+
+  // clone the runtimeConfig from existingRuntime
+  const clonedRuntimeConfig = cloneDeep(existingRuntime.runtimeConfig);
+
+  // check if identity is defined in the runtimeConfig
+  if (!clonedRuntimeConfig.identity) {
```

Comment:
What happens, when in the initial runtime, a identity parameter was passed to? Then i will get the runtime for the initial identity again, or?